### PR TITLE
Drop support for Python 3.7

### PR DESCRIPTION
**Pull request by srittau**
Part of #10113

### CODE REVIEW

**Comment by srittau** [comment]
This PR contains only infrastructure work. Changes to the stubs themselves will follow in a subsequent PR.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,16 +5,15 @@ range of Python users and Python codebases. If you're trying a type
 checker on your Python code, your experience and what you can contribute
 are important to the project's success.
 
-
```
Comment (JelleZijlstra) [inline CONTRIBUTING.md:8]: Maybe remove the formatting changes?

**Review by JelleZijlstra — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (srittau) [reply CONTRIBUTING.md:8]: VSCode is driving me crazy ...

**Comment by AlexWaygood** [comment]
You can also delete the lint in `check_new_syntax.py` that checks we're not using any PEP-570 syntax (but it's also fine to leave that to a followup PR)

**Comment by AlexWaygood** [comment]
And this can also be deleted: https://github.com/python/typeshed/blob/54447f63a4c9445de67d6708368a3e009713ca36/CONTRIBUTING.md?plain=1#L282-L286

**Comment by srittau** [comment]
> And this can also be deleted:
> 
> https://github.com/python/typeshed/blob/54447f63a4c9445de67d6708368a3e009713ca36/CONTRIBUTING.md?plain=1#L282-L286

I removed it, but didn't add it to supported features, as it's not really a typing feature.

**Review by AlexWaygood — APPROVED** [review]
Thanks!

**Comment by robsdedude** [comment]
Sorry for pinging on such an old issue, but I'd like to bring to your attention for the next Python version drop: it's quite difficult for a given stub package to find out in which package version which python version was dropped support for. This is relevant for packages trying to keep support for older Python versions.

**Pull request merged**